typings: getForRepo returns Promise<Github.AnyResponse> instead of a promise for an iterable

I used to be able to do

```
for (const issue of await octokit.issues.getForRepo({ owner, repo, state: 'open', labels: 'translation' })) {
...
}
```

but now I get

```
error TS2488: Type must have a '[Symbol.iterator]()' method that returns an iterator.
```

--- a/package.json
+++ b/package.json
@@ -72,7 +72,7 @@
     "build:flow": "node scripts/generate-flow-types",
     "build:ts": "node scripts/generate-typescript-types",
     "prevalidate:ts": "npm run -s build:ts",
-    "validate:ts": "tsc index.d.ts",
+    "validate:ts": "tsc --target es6 index.d.ts",
     "deploy-docs": "gh-pages-with-token -d doc",
     "semantic-release": "semantic-release"
   },

--- a/scripts/templates/index.d.ts.tpl
+++ b/scripts/templates/index.d.ts.tpl
@@ -22,7 +22,7 @@ declare namespace Github {
       etag: string,
       status: string
     }
-    
+
     [Symbol.iterator](): Iterator<any>
   }
 

--- a/scripts/templates/index.d.ts.tpl
+++ b/scripts/templates/index.d.ts.tpl
@@ -22,6 +22,8 @@ declare namespace Github {
       etag: string,
       status: string
     }
+    
+    [Symbol.iterator](): Iterator<any>
   }
 
   export interface EmptyParams {

